### PR TITLE
add support for v2 basemap styles

### PR DESCRIPTION
**Pull request by gowin20**
Adds support for the [basemap styles service v2 beta](https://developers.arcgis.com/rest/basemap-styles/) as mentioned in #174 

* New `getBasemapStyleV2Url` function that returns a reference to the new v2 endpoint
* New `version` parameter in `VectorBasemapLayer` that specifies the endpoint to access (1 or 2, 1 by default)
* New `language` parameter that works with v2 of the service, accepts valid [language parameters](https://developers.arcgis.com/rest/basemap-styles/arcgis-outdoor-style-get/#language)

### CODE REVIEW

**Review by gavinr — CHANGES_REQUESTED** [review]
Overall looks solid. Here are some ideas, and also can you add some unit tests to `VectorBasemapLayerSpec.js` to test this new functionality? Thanks!

File: examples/languages.html
Comment (gowin20) [reply examples/languages.html:96]: It is currently static. I asked the team if they had any plans to add an endpoint to request a list of supported languages. Not sure if that will be added, we'll have to wait and see.

**Review by gavinr — APPROVED** [review]
Looks good! Thanks for adding the tests. Waiting for @patrickarlt to review before merge.

**Comment by gowin20** [comment]
included in #182

**Pull request closed without merging**
